### PR TITLE
added new pressure sensor

### PR DESCRIPTION
**Pull request by JonBruchim**
Signed-off-by: rinslow <yonbruchim@gmail.com>

### CODE REVIEW

File: drivers/abp_pressure_sensor.py [hunk 1]
```diff
@@ -1,46 +1,22 @@
-import pigpio
 import logging
 
-from errors import I2CReadError
-
-from .i2c_driver import I2cDriver
+from honeywell_pressure_sensor import HoneywellPressureSensor
```
Comment (osherdp) [inline drivers/abp_pressure_sensor.py:3]: use relative import:
```suggestion
from .honeywell_pressure_sensor import HoneywellPressureSensor
```

File: drivers/honeywell_pressure_sensor.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    MEASURE_BYTE_COUNT = 0x2
+    MAX_RANGE_PRESSURE = NotImplemented
+    MIN_RANGE_PRESSURE = NotImplemented
+    MAX_OUT_PRESSURE = NotImplemented
+    MIN_OUT_PRESSURE = NotImplemented
+    SENSITIVITY = NotImplemented
+    CMH2O_RATIO = NotImplemented
+
+    def __init__(self):
+        super().__init__()
```
Comment (osherdp) [inline drivers/honeywell_pressure_sensor.py:23]: you can delete

File: drivers/honeywell_pressure_sensor.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    def _calculate_pressure(self, pressure_reading):
+        pressure = (((pressure_reading - self.MIN_OUT_PRESSURE) *
+                    self.SENSITIVITY) + self.MIN_RANGE_PRESSURE)
```
Comment (osherdp) [inline drivers/honeywell_pressure_sensor.py:27]: That's a mess
```suggestion
        pressure = (self.MIN_RANGE_PRESSURE +
                    self.SENSITIVITY * (pressure_reading - self.MIN_OUT_PRESSURE))
```

File: drivers/honeywell_pressure_sensor.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        cmh2o_pressure = pressure * self.CMH2O_RATIO
+        return cmh2o_pressure
+
+    def read(self):
+        """ Returns pressure as cmh2o """
+        try:
+            read_size, pressure_raw = self._pig.i2c_read_device(self._dev, self.MEASURE_BYTE_COUNT)
+            if read_size >= self.MEASURE_BYTE_COUNT:
+                pressure_reading = ((pressure_raw[0] & 0x3F) << 8) | (pressure_raw[1])
+                return (self._calculate_pressure(pressure_reading))
```
Comment (osherdp) [inline drivers/honeywell_pressure_sensor.py:37]: remove redundant parenthesis

File: drivers/honeywell_pressure_sensor.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            else:
+                # Todo: Do we need to retry reading after a little sleep? (see flow sensor logic)
```
Comment (osherdp) [inline drivers/honeywell_pressure_sensor.py:39]: why not just doing it in the next iteration?

File: drivers/honeywell_pressure_sensor.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                log.error("Pressure sensor's measure data not ready")
```
Comment (osherdp) [inline drivers/honeywell_pressure_sensor.py:40]: warning?

File: drivers/honeywell_pressure_sensor.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        except pigpio.error as e:
+            log.error("Could not read from pressure sensor. "
+                      "Is the pressure sensor connected?.")
+            raise I2CReadError("i2c write failed") from e
```
Comment (osherdp) [inline drivers/honeywell_pressure_sensor.py:44]: Can you make sure we're not crashing if the sensor is not connected?

File: drivers/abp_pressure_sensor.py [hunk 8]
```diff
@@ -1,46 +1,22 @@
...
 
 log = logging.getLogger(__name__)
 
 
-class AbpPressureSensor(I2cDriver):
+class AbpPressureSensor(HoneywellPressureSensor):
```
Comment (osherdp) [inline drivers/abp_pressure_sensor.py:8]: What's the difference between this driver and the below one?
They have the same class name

Comment (JonBruchim) [reply drivers/abp_pressure_sensor.py:8]: Its the same class , now all honeywell pressure sensors derive from the same base class.
Honeywell pressure sensor

File: drivers/hsc_pressure_sensor.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    def _pressure_to_flow(self, pressure_cmh2o):
+        flow = (abs(pressure_cmh2o) ** 0.5) * self.SYSTEM_RATIO
+        if pressure_cmh2o < 0:
+            flow = -flow
+
+        return flow
+
+    def read(self):
+        dp_cmh2o = super(HscPressureSensor, self).read()
+        print(dp_cmh2o)
```
Comment (osherdp) [inline drivers/hsc_pressure_sensor.py:35]: delete

File: drivers/hsc_pressure_sensor.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    SYSTEM_RATIO = 36.609
+
+    def __init__(self):
+        super().__init__()
+        log.info("HSC pressure sensor initialized")
+
+    def _pressure_to_flow(self, pressure_cmh2o):
+        flow = (abs(pressure_cmh2o) ** 0.5) * self.SYSTEM_RATIO
+        if pressure_cmh2o < 0:
+            flow = -flow
```
Comment (osherdp) [inline drivers/hsc_pressure_sensor.py:29]: Can you explain

File: drivers/driver_factory.py [hunk 11]
```diff
@@ -171,8 +171,8 @@ def get_pressure_driver():
 
     @staticmethod
     def get_flow_driver():
-        from drivers.sfm3200_flow_sensor import Sfm3200
-        return Sfm3200()
+        from drivers.hsc_pressure_sensor import HscPressureSensor
+        return HscPressureSensor()
```
Comment (osherdp) [inline drivers/driver_factory.py:175]: you sure it's the same?

**Pull request merged**